### DISCUSSION TITLE
Macro-Generated Field Referencing

### DISCUSSION THREAD

**Discussion by Zeimyth** [scala/scala3#13073] [Metaprogramming] [answered]
With Scala 2 macros, it was possible to take an arbitrary string and use it as the name of a field to reference:

```scala
val term = TermName("foo")
q"obj.$term"
```

Is this possible with the reflection API in Scala 3 macros? I feel like `Symbol`s might be the most likely place for me to find what I'm looking for, but a `Symbol` seems to point me to the definition of an object's class. I can look up methods and fields defined on the class from a `Symbol`, but I can't figure out how to turn that into a field reference on an instance of that class.

**Comment by rayrobdod** [comment, marked as answer]
I don't know if there is a way to splice these into quasiquotes. When working with reflect trees, a `Term` can represent an instance of a class, and a `Symbol` can represent a method or field name. If you have a `Term` representing an object (probably from `Expr#asTerm`) and a `Symbol` representing the field, you can create a tree that calls the field on the object by creating a `Select` tree, either with `Term#select` or one of the `SelectModule` methods. Then the `Select` tree can be turned back into an `Expr` with `Term#asExprOf`.

**Reply by Zeimyth** [reply]
Worked like a charm! `Term#select` was the missing piece. Thank you!
